packages/backend-defaults/report-httpAuth.api.md

```diff
@@ -3,9 +3,50 @@
 > Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
 
 ```ts
+import { AuthService } from '@backstage/backend-plugin-api';
+import { BackstageCredentials } from '@backstage/backend-plugin-api';
+import { BackstagePrincipalTypes } from '@backstage/backend-plugin-api';
+import { DiscoveryService } from '@backstage/backend-plugin-api';
 import { HttpAuthService } from '@backstage/backend-plugin-api';
+import { Request as Request_2 } from 'express';
+import { Response as Response_2 } from 'express';
 import { ServiceFactory } from '@backstage/backend-plugin-api';
 
+// @public
+export class DefaultHttpAuthService implements HttpAuthService {
+  // (undocumented)
+  static create(options: DefaultHttpAuthServiceOptions): DefaultHttpAuthService;
+  // (undocumented)
+  credentials<TAllowed extends keyof BackstagePrincipalTypes = 'unknown'>(
+    req: Request_2,
+    options?: {
+      allow?: Array<TAllowed>;
+      allowLimitedAccess?: boolean;
+    },
+  ): Promise<BackstageCredentials<BackstagePrincipalTypes[TAllowed]>>;
+  // (undocumented)
+  issueUserCookie(
+    res: Response_2,
+    options?: {
+      credentials?: BackstageCredentials;
+    },
+  ): Promise<{
+    expiresAt: Date;
+  }>;
+}
+
+// @public
+export interface DefaultHttpAuthServiceOptions {
+  // (undocumented)
+  auth: AuthService;
+  // (undocumented)
+  discovery: DiscoveryService;
+  // (undocumented)
+  extractTokenFromRequest?: (req: Request_2) => string | undefined;
```

Comment:
Maybe make it an object as the return type, that way we can put more contextual info in there if needed (what if it's actually not easy to know JUST by looking at the token value what service it originated from? for non-jwks etc)